File: benchmark/cases/prime_sieve/prime_sieve.py
import os
import time


def prime_sieve(limit: int) -> None:
    print("primes up to", limit)
    count = 0
    n = 2
    while n <= limit:
        is_prime = True
        d = 2
        while d < n:
            if n % d == 0:
                is_prime = False
                break
            d += 1
        if is_prime:
            print("  prime:", n)
            count += 1
        n += 1
    print("total primes up to", limit, "=", count)


t0 = time.perf_counter_ns()
prime_sieve(int(os.environ["BENCH_SCALE"]))
t1 = time.perf_counter_ns()
print("__bench_ns:", t1 - t0)

File: benchmark/cases/prime_sieve/prime_sieve.rs
use std::time::Instant;

fn prime_sieve(limit: i64) {
    println!("primes up to {}", limit);
    let mut count = 0;
    let mut n = 2;
    while n <= limit {
        let mut is_prime = true;
        let mut d = 2;
        while d < n {
            if n % d == 0 {
                is_prime = false;
                break;
            }
            d += 1;
        }
        if is_prime {
            println!("  prime: {}", n);
            count += 1;
        }
        n += 1;
    }
    println!("total primes up to {} = {}", limit, count);
}

fn main() {
    let scale: i64 = std::env::var("BENCH_SCALE").unwrap().parse().unwrap();
    let t0 = Instant::now();
    prime_sieve(scale);
    let ns = t0.elapsed().as_nanos();
    println!("__bench_ns: {}", ns);
}
